File: external_apps/mobile_ios/mobile_ios/Core/Model/PaymentModels.swift
//
//  PaymentModels.swift
//  mobile_ios
//
//  Created for Payment & Billing Feature
//

import Foundation

enum TransactionStatus: String, Codable {
    case pending = "PENDING"
    case success = "SUCCESS"
    case failed = "FAILED"
    case refunded = "REFUNDED"
}

struct Transaction: Identifiable, Codable {
    let id: String
    let amount: Double
    let currency: String
    let status: TransactionStatus
    let method: String // Stripe, VNPay, MoMo, IAP
    let date: Date
    let description: String
}

struct Invoice: Identifiable, Codable {
    let id: String
    let transactionId: String
    let url: String // URL to PDF
    let issuedDate: Date
    let customerName: String
    let items: [InvoiceItem]
}

struct InvoiceItem: Identifiable, Codable {
    let id = UUID()
    let description: String
    let quantity: Int
    let unitPrice: Double
}

struct StripePaymentConfig: Codable {
    let paymentIntentClientSecret: String
    let ephemeralKeySecret: String
    let customerId: String
    let publishableKey: String
}

File: external_apps/mobile_android/app/src/main/java/com/example/mobile_android/data/model/PaymentModels.kt
package com.example.mobile_android.data.model

import java.util.Date

data class Transaction(
    val id: String,
    val amount: Double,
    val currency: String,
    val status: TransactionStatus,
    val method: String, // Stripe, VNPay, MoMo, IAP
    val date: Date,
    val description: String
)

enum class TransactionStatus {
    PENDING, SUCCESS, FAILED, REFUNDED
}

data class Invoice(
    val id: String,
    val transactionId: String,
    val url: String, // URL to PDF
    val issuedDate: Date,
    val customerName: String,
    val items: List<InvoiceItem>
)

data class InvoiceItem(
    val description: String,
    val quantity: Int,
    val unitPrice: Double
)

data class StripePaymentConfig(
    val paymentIntentClientSecret: String,
    val ephemeralKeySecret: String,
    val customerId: String,
    val publishableKey: String
)
